bin/compile

```diff
@@ -62,28 +61,68 @@ export_env_dir() {
 
 function copy_binaries_from_cache() {
   echo "Copying tailscale binaries to $BUILD_DIR/bin" | indent
-  mkdir -p "$BUILD_DIR"/bin
+  mkdir -p "$BUILD_DIR/bin"
   cp "$CACHE_DIR/tailscale" "$BUILD_DIR/bin/tailscale"
   cp "$CACHE_DIR/tailscaled" "$BUILD_DIR/bin/tailscaled"
 }
 
 function copy_binaries_to_cache() {
-  cp /usr/bin/tailscale "$CACHE_DIR/tailscale"
-  cp /usr/sbin/tailscaled "$CACHE_DIR/tailscaled"
+  cp "$APT_CACHE_DIR/tailscale/usr/bin/tailscale" "$CACHE_DIR/tailscale"
+  cp "$APT_CACHE_DIR/tailscale/usr/sbin/tailscaled" "$CACHE_DIR/tailscaled"
 }
 
 function install_from_apt() {
   # load the VERSION_CODENAME ENV var from the OS.
   . /etc/os-release
 
+  APT_CACHE_DIR="$HOME/apt/cache"
```

Comment:
All of these variables as well as all of the directory creating and permissions management below are all to make sure we have a location on the filesystem that whatever user we're logged in as (`root` on `heroku-22` and below, and `heroku` on `heroku-24` and above) has permissions to write to.

There are still a few warnings that pop up when running this script, but so far they do not seem to be an actual issue, in that, we can still install Tailscale successfully, and because the Tailscale binaries are self contained, once we install them using `dpkg` and copy the installed files to our `$CACHE_DIR` nothing else about the `apt` environment is required.

Thank goodness for compiled `go` binaries, I guess. 😄